### ISSUE TITLE
Log4J Upgrade

### ISSUE THREAD

**Issue by ex3ndr** [usnistgov/jsip#6]
Hey!

log4j 1.2 is already reached end-of-life and isn't jsip should be updated to log4j 2?

Thanks.

**Comment by vladimirralev** [comment]
Right. It's probably best to drop the log4j binary dependency and just have the logging backend open in some back-compatible way.

**Comment by ranganathanm** [comment]
log4j sounds like some friendly old ghost :-)  -- its been a long time since I gave it any thought.


I thought we removed the log4j dependency at some point (the default logger is log4j but IIRC, we removed that dependency.)


Ranga.

________________________________

**Comment by livem** [comment]
I think slf4j is the best choice. http://www.slf4j.org

**Comment by livem** [comment]
I think **slf4j** is the best choice. http://www.slf4j.org

**Comment by ganeshbms** [comment]
one request, please make debug and server logs rollable. I see debug log file size growing to GBs if I keep the debug level under heavy load.

I have upgraded my application to latest Log4J 2.x and observed that, sip stack still logs messages to debug and server log files but without considering logging format defined by log4J 2.x log config file.
Looks like, some compliance issues.

Thanks,
Ganesh
